bin/setup.bat

```diff
@@ -15,6 +15,13 @@ goto finally
 
 :setup_jruby
 REM setup_java()
+if not defined JAVA_HOME IF EXIST %ProgramData%\Oracle\java\javapath\java.exe (
+    for /f "tokens=2 delims=[]" %%a in ('dir %ProgramData%\Oracle\java\javapath\java.exe') do @set JAVA_EXE=%%a
+)
+if defined JAVA_EXE set JAVA_HOME=%JAVA_EXE:\bin\java.exe=%
```

Comment:
@suyograo No problem. I thought about my solution and find that a temporary variable JAVA_EXE fits better here. So I made the last commit. This for loop sets eg. JAVA_EXE = C:\Program Files\Java\jdk1.6.0_02\bin\java.exe as it only can read the complete symlink to the java.exe binary file.

Here in line 21 the JAVA_HOME variable is then set with the value of JAVA_EXE, but without the \bin\java.exe, so we set here JAVA_HOME = C:\Program Files\Java\jdk1.6.0_02

The code works as it is needed for a correct JAVA_HOME env in the script, I've tested this script on a machine with Java 8 installed, but no user/system JAVA_HOME environment set.